File: ga/4omini_result_2024_7/result/1968B_mini/43_1968B.c
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut lines = reader.lines();

    let t: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();

    for _ in 0..t {
        let first_line = lines.next().unwrap().unwrap();
        let parts: Vec<&str> = first_line.split_whitespace().collect();
        
        let n: usize = parts[0].parse().unwrap();
        let m: usize = parts[1].parse().unwrap();
        
        let sn = lines.next().unwrap().unwrap();
        let sm = lines.next().unwrap().unwrap();
        
        let (mut i, mut j) = (0, 0);
        
        while i < n && j < m {
            if sn.chars().nth(i).unwrap() == sm.chars().nth(j).unwrap() {
                i += 1;
            }
            j += 1;
        }
        
        println!("{}", i);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968B_gpt/43_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: i32 = input.trim().parse().unwrap();

    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let inputs: Vec<i32> = input.trim().split(" ").map(|x| x.parse().unwrap()).collect();
        let n = inputs[0];
        let m = inputs[1];

        let mut sn = String::new();
        io::stdin().read_line(&mut sn).unwrap();
        let sn = sn.trim();

        let mut sm = String::new();
        io::stdin().read_line(&mut sm).unwrap();
        let sm = sm.trim();

        let mut i = 0;
        let mut j = 0;

        while i < n && j < m {
            if sn.chars().nth(i as usize) == sm.chars().nth(j as usize) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}